src/components/views/rooms/RoomSublist2.tsx

```diff
@@ -109,6 +109,11 @@ export default class RoomSublist2 extends React.Component<IProps, IState> {
         this.forceUpdate(); // because the layout doesn't trigger a re-render
     };
 
+    private onShowLessClick = () => {
+        this.props.layout.visibleTiles = this.props.layout.minVisibleTiles;
+        this.forceUpdate(); // because the layout doesn't trigger a re-render
```

Comment:
This seems like a weird anti-pattern to introduce in a refactor: is this a case for a split-out view model type thing? It's not clear which components are & aren't responsible for updating this object.

Reply:
a view model could certainly help, though this isn't the first place `forceUpdate()` is used in this refactor.

Reply:
yeah, fair enough